pennylane/workflow/qnode.py

```diff
@@ -512,6 +512,8 @@ def __init__(
         mcm_method: Literal[None, "deferred", "one-shot", "tree-traversal"] = None,
         **gradient_kwargs,
     ):
+        self.init_args = locals()
+        del self.init_args["self"]

```

Comment:
Probably best to make this a protected attribute, i.e `_init_args` instead of `init_args`